parachain/primitives/router/src/outbound/mod.rs

```diff
@@ -197,7 +200,7 @@ impl<'a, Call> XcmConverter<'a, Call> {
 		let max_target_fee = self.fee_info()?;
 
 		// Get withdraw/deposit and make native tokens create message.
-		let result = self.native_tokens_unlock_message()?;
+		let result = self.agent_execute_message()?;
```

Comment:
Nice! And just realize that the original xcm transact should be constructed from another parachain and sent through a `BridgeXcmSender`(e.g.  take [BridgeXcmSender](https://github.com/Snowfork/cumulus/blob/5f9f6cd8eb6d779d8b713dd7f931a9521e93b132/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs#L727) for `pallet_bridge_transfer` in `asset_hub_runtime` as reference).

Since a template runtime is introduced in https://github.com/Snowfork/cumulus/pull/56,  we can wrap a transact extrinsic into it and then do some e2e tests accordingly.

Reply:
Currently `asset_hub_runtime` use [UnpaidRemoteExporter](https://github.com/Snowfork/cumulus/blob/5f9f6cd8eb6d779d8b713dd7f931a9521e93b132/parachains/runtimes/assets/asset-hub-kusama/src/xcm_config.rs#L635-L636)  for fee of  asset transfer.  But when it comes to arbitrary message from arbitrary sibling parachain,  [SovereignPaidRemoteExporter](https://github.com/paritytech/polkadot/blob/8f05479e4bd61341af69f0721e617f01cbad8bb2/xcm/xcm-builder/src/universal_exports.rs#L217) make more sense and the delivery fee [here] should also be carefully addressed.

Reply:
> Nice! And just realize that the original xcm transact should be constructed from another parachain and sent through a BridgeXcmSender(e.g. take [BridgeXcmSender](https://github.com/Snowfork/cumulus/blob/5f9f6cd8eb6d779d8b713dd7f931a9521e93b132/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs#L727) for pallet_bridge_transfer in asset_hub_runtime as reference).

Should `BridgeXcmSender` be added to the bridge hub runtime? Or to the template pallet runtime?